addOns/automation/src/test/java/org/zaproxy/addon/automation/jobs/ExitStatusJobUnitTest.java

```diff
@@ -348,22 +349,64 @@ void shouldSetExitCode(String alertrisk, String exitcode) {
         assertThat(ExtensionAutomation.getExitOverride(), is(equalTo(Integer.parseInt(exitcode))));
     }
 
-    private JobResultData getTestData(String alertLevel) {
-        Alert alert = new Alert(-1, JobUtils.parseAlertRisk(alertLevel), 2, "test");
+    @ParameterizedTest
+    @CsvSource({
+        "HIGH,MEDIUM,4",
+        "medium,medium,3",
+        "low,medium,2",
+        "High,False Positive,0",
+        "Medium,False Positive,0",
+        "Low,False Positive,0"
+    })
+    void shouldSetExitCodeExcludingFalsePositive(
+            String alertrisk, String confidence, String exitcode) {
+        // Given
+        ExitStatusJob job = new ExitStatusJob();
+        AutomationProgress progress = new AutomationProgress();
+        progress.addJobResultData(getTestData(alertrisk, confidence));
+
+        // When
+        job.getParameters().setOkExitValue(Integer.parseInt(exitcode) > 0 ? 2 : 0);
+        job.getParameters().setWarnExitValue(3);
+        job.getParameters().setErrorExitValue(4);
+        job.getParameters().setErrorLevel("high");
+        job.getParameters().setWarnLevel("medium");
+        job.verifyParameters(progress);
+        job.runJob(new AutomationEnvironment(progress), progress);
 
-        JobResultData data =
-                new JobResultData("test") {
+        // Then
+        Collection<JobResultData> data = progress.getAllJobResultData();
+        Collection<Alert> alerts = new ArrayList<>();
+        data.forEach(e -> alerts.addAll(e.getAllAlertData()));
+        assertThat(alerts.size(), is(equalTo(1)));
+        Alert alert = (Alert) ((ArrayList<?>) alerts).get(0);
+        assertThat(alert.getConfidence(), is(equalTo(JobUtils.parseAlertConfidence(confidence))));
```

Comment:
Why?

Reply:
To ensure it's the one we think it is. Could remove it.

Reply:
Aren't we creating it in the Given with the expected confidence?